Automattic/jetpack
# Social | Hide "Mark as shared" for non-admin authors

For non-admin authors, "Mark the connection as shared" checkbox is visible, which should not be the case.

## Proposed changes:

* Hide "Mark as shared" for non-admin authors

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






- Ensure that feature flag is ON - `define( 'JETPACK_SOCIAL_USE_ADMIN_UI_V1', true );`
- Test the following for both Jetpack and Social plugin
    - Goto connections management UI
    - Add a connection
    - Create a non-admin author account
    - Login as the author
    - Create a post
    - Open the Jetpack/Social sidebar and open the connections modal
    - Add a connection
    - On the connection confirmation screen, confirm that you don't see the "Mark as shared..." checkbox
    - Confirm the connection
    - Open the modal again
    - Confirm that the added connection does not show "Mark as shared"
    - Confirm that everything else works as expected for admin account

<table>
    <tbody>
        <tr>
            <th>BEFORE</th>
            <th>AFTER</th>
        </tr>
        <tr>
            <th colspan="2">Confirmation screen</th>
        </tr>
        <tr>
            <td>
<img width="1051" alt="Screenshot 2024-05-28 at 2 38 08 PM" src="https://github.com/Automattic/jetpack-reach/assets/18226415/fe13425a-321a-4f71-9699-7b9d693ae5b0">
</td>
            <td>
<img width="1051" alt="Screenshot 2024-05-28 at 2 38 08 PM" src="https://github.com/Automattic/jetpack/assets/18226415/88179ca5-3df9-4517-b25f-b1cbfb6cf235">
</td>
        </tr>
        <tr>
            <th colspan="2">Inside modal</th>
        </tr>
        <tr>
            <td>
<img width="1016" alt="Screenshot 2024-05-28 at 2 41 06 PM" src="https://github.com/Automattic/jetpack-reach/assets/18226415/90e14e0c-e92f-4287-9a55-bffb7504b012">
</td>
            <td>
<img width="1016" alt="Screenshot 2024-05-28 at 2 41 06 PM" src="https://github.com/Automattic/jetpack/assets/18226415/302096ee-01e1-4ebe-bec3-aa55df39ea70">
</td>
        </tr>
    </tbody>
</table>
